editor/src/components/editor/global-shortcuts.ts

```diff
@@ -110,11 +110,21 @@ import {
   TOGGLE_FOCUSED_OMNIBOX_TAB,
   FOCUS_CLASS_NAME_INPUT,
 } from './shortcut-definitions'
-import { DerivedState, EditorState, getOpenFile } from './store/editor-state'
+import { DerivedState, EditorState, getOpenFile, RightMenuTab } from './store/editor-state'
 import { CanvasMousePositionRaw, WindowMousePositionRaw } from '../../utils/global-positions'
 import { getDragStateStart } from '../canvas/canvas-utils'
 import { isFeatureEnabled } from '../../utils/feature-switches'
 
+function cancelInsertModeActions(): Array<EditorAction> {
+  return [
+    EditorActions.switchEditorMode(EditorModes.selectMode()),
+    CanvasActions.clearDragState(false),
+    EditorActions.clearHighlightedViews(),
+    CanvasActions.clearInteractionSession(false),
+    EditorActions.setRightMenuTab(RightMenuTab.Inspector),
+  ]
```

Comment:
It seems we could use this function here too: https://github.com/concrete-utopia/utopia/blob/master/editor/src/templates/editor-canvas.tsx#L188
And maybe here too: https://github.com/concrete-utopia/utopia/blob/master/editor/src/components/canvas/controls/insert-mode-control-container.tsx#L432

Reply:
There's a subsequent task on the list which involves another case for cancelling insert mode, I think it makes sense to do that refactoring there as my first attempt resulted in circular references.